src/game_main.cpp

```diff
@@ -162,6 +163,28 @@ static std::string findIntroLevel()
     return selected;
 }
 
+// expand the section vertically if the top 8px of the level are empty
+static void s_ExpandSectionForMenu()
+{
+    // check current section top for expandability
+    Location_t tempLocation = newLoc(level[Player[1].Section].X, level[0].Y, level[Player[1].Section].Width - level[Player[1].Section].X, 8);
+
+    for(int A = 1; A <= numBlock; A++)
+    {
+        if(CheckCollision(Block[A].Location, tempLocation))
+            return;
+    }
+
+    for(int A = 1; A <= numBackground; A++)
+    {
+        if(CheckCollision(Background[A].Location, tempLocation))
+            return;
+    }
+
+    // expand level height to a maximum of 2160px
+    if(level[Player[1].Section].Y > level[Player[1].Section].Height - 2160)
+        level[Player[1].Section].Y = level[Player[1].Section].Height - 2160;
+}
```

Comment:
Please avoid so many complex deep expressions repeated, please make the reference:
```cpp
auto &section = level[Player[1].Section];
```
and then, re-use the `section` reference. This thing will be easier to maintain the logic.

Reply:
Better still, I see now that some code mixing happened here. It's all meant to be `level[0]`, though they are the same on any valid `intro.lvl` files. I'll fix this. Thanks for the note!